### PR TITLE
SASS-7117: Implemented TYS API 1903

### PR DESCRIPTION
**Pull request by ansahdavid**
### Description
Added code and tests to implement TYS API 1903

Work related to this PR is for ticket SASS-7116

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/services/CreateOrAmendSavingsService.scala [hunk 1]
```diff
@@ -16,21 +16,30 @@
...
+   createOrAmendSavingsConnector: CreateOrAmendSavingsConnector,
+   createOrAmendSavingsTysConnector: CreateOrAmendSavingsTysConnector
+) extends Logging {
 
   def createOrAmendSavings(nino: String, taxYear: Int, savingsModel: CreateOrAmendSavingsModel)
-                          (implicit hc: HeaderCarrier): Future[CreateOrAmendSavingsResponse] =
-    createOrAmendSavingsConnector.createOrAmendSavings(nino, taxYear, savingsModel)
+                          (implicit hc: HeaderCarrier): Future[CreateOrAmendSavingsResponse] = {
+
+    if(taxYear < specificTaxYear) {
```
Comment (basotia) [inline app/services/CreateOrAmendSavingsService.scala:39]: check with other repo

File: it/connectors/CreateOrAmendSavingsTysConnectorISpec.scala [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  )
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+  lazy val connector: CreateOrAmendSavingsTysConnector = app.injector.instanceOf[CreateOrAmendSavingsTysConnector]
+  lazy val httpClient: HttpClient = app.injector.instanceOf[HttpClient]
+
+  def mkTaxYear(taxYear: Int): String = s"${taxYear - 1}-${taxYear.toString takeRight 2}"
+
+  val nino = "nino"
+  val taxYear = 2024
```
Comment (basotia) [inline it/connectors/CreateOrAmendSavingsTysConnectorISpec.scala:45]: read from util

Comment (ansahdavid) [reply it/connectors/CreateOrAmendSavingsTysConnectorISpec.scala:45]: Yep, made this change in the next commit

File: app/services/CreateOrAmendSavingsService.scala (hunk 1 above)
Comment (ansahdavid) [reply app/services/CreateOrAmendSavingsService.scala:39]: Checked with other repo's and have aligned this logic to how it has previously been done for consistency in the next commit

**Review by basotia — APPROVED** [review]

**Pull request merged**